Scripts/MapLoaders/MinimapHook.as

```diff
@@ -1,96 +1,76 @@
+	
+	const SColor color_minimap_castle_back  (0xff313412);
+	const SColor color_minimap_wood_back    (0xff552A11);
+	
 	const SColor color_minimap_water        (0xff2cafde);
 	const SColor color_minimap_fire         (0xffd5543f);
-
-	//neighbours
-	Tile tile_l = map.getTile(MiniMap::clampInsideMap(pos * ts - Vec2f(ts, 0), map));
-	Tile tile_r = map.getTile(MiniMap::clampInsideMap(pos * ts + Vec2f(ts, 0), map));
-	Tile tile_u = map.getTile(MiniMap::clampInsideMap(pos * ts - Vec2f(0, ts), map));
-	Tile tile_d = map.getTile(MiniMap::clampInsideMap(pos * ts + Vec2f(0, ts), map));
-
-	///figure out the correct colour
-	if (
-		//always solid
-		map.isTileGround( tile ) || map.isTileStone( tile ) ||
-        map.isTileBedrock( tile ) || map.isTileThickStone( tile ) ||
-        map.isTileCastle( tile ) || map.isTileWood( tile ) ||
-        //only solid if we're not showing gold separately
-        (!show_gold && map.isTileGold( tile ))
-    ) {
-		//Foreground
-		col = color_minimap_solid;
-
-		//Edge
-		if( MiniMap::isForegroundOutlineTile(tile_u, map) || MiniMap::isForegroundOutlineTile(tile_d, map) ||
-		    MiniMap::isForegroundOutlineTile(tile_l, map) || MiniMap::isForegroundOutlineTile(tile_r, map) )
-		{
-			col = color_minimap_solid_edge;
+	
+	// bool isTileInFire(int xTilespace, int yTilespace)
+	// bool isTileSolid(uint16 tile)
+	// bool isTileSolid(const Tile&in tile)
+	// bool isTileSolid(Vec2f posWorldspace)
+	// bool isTileCollapsing(uint offset)
+	// bool isTileCollapsing(Vec2f posWorldspace)
+	// bool isTileLadder(const Tile&in tile)
+	// bool isTileBackground(const Tile&in tile)
+	// bool isTileBackgroundNonEmpty(const Tile&in tile)
+	// bool isTileGroundBack(uint16 tile)
+	// bool isTileGold(uint16 tile)
+	// bool isTileGrass(uint16 tile)
+	// bool isTileWood(uint16 tile)
+	// bool isTileCastle(uint16 tile)
+	// bool isTileSand(uint16 tile)
+	// bool isTileGroundStuff(uint16 tile)
+	// bool isTilePlatform(const Tile&in tile)
+	
+	if (show_gold && map.isTileGold(tile))  { col = color_minimap_gold;
+	} else if (map.isTileGround(tile))      { col = color_minimap_ground;
+	} else if (map.isTileThickStone(tile))  { col = color_minimap_thickstone;
+	} else if (map.isTileStone(tile))       { col = color_minimap_stone;
+	} else if (map.isTileBedrock(tile))     { col = color_minimap_bedrock;
+	} else if (map.isTileWood(tile))        { col = color_minimap_wood;
+	} else if (map.isTileCastle(tile))      { col = color_minimap_castle;
+	
+	// } else if(map.isTileBackground(ctile) && !map.isTileGrass(tile))
+	} else if(map.isTileBackgroundNonEmpty(ctile) && !map.isTileGrass(tile)) {
+		
+		// TODO(hobey): maybe check if there's a door/platform on this backwall and make a custom color for them?
+		if        (tile == CMap::tile_castle_back) {
+			col =       color_minimap_castle_back;
+		} else if (tile == CMap::tile_wood_back)   {
+			col =       color_minimap_wood_back;
+		} else {
+			col = color_minimap_back;
```

Comment:
This entire section needs to be formatted better:
* Put curly brackets on their own line
* Add a space after `if` on line 59
* Remove excess whitespace after `if` on line 62